omnigibson/__init__.py

```diff
@@ -45,6 +45,79 @@
 tempdir = tempfile.mkdtemp()
 
 
+def clear():
+    """
+    Clear the stage and then call launch_simulator again to make og.sim point to a new simulator instance
+    """
+    global sim
+
+    import omnigibson.lazy as lazy
+    from omnigibson.object_states.update_state_mixin import GlobalUpdateStateMixin
+    from omnigibson.prims.material_prim import MaterialPrim
+    from omnigibson.sensors.vision_sensor import VisionSensor
+    from omnigibson.transition_rules import TransitionRuleAPI
+    from omnigibson.utils.python_utils import clear as clear_python_utils
+    from omnigibson.utils.usd_utils import clear as clear_usd_utils
+
+    # Stop the physics
+    sim.stop()
+
+    # Clear all scenes
+    for scene in sim.scenes:
+        scene.clear()
+    sim._scenes = []
+
+    # Remove the skybox, floor plane and viewer camera
+    if sim._skybox is not None:
+        sim._skybox.remove()
+        sim._skybox = None
+
+    if sim._floor_plane is not None:
+        sim._floor_plane.remove()
+        sim._floor_plane = None
+
+    if sim._viewer_camera is not None:
+        sim._viewer_camera.remove()
+        sim._viewer_camera = None
+
+    if sim._camera_mover is not None:
+        sim._camera_mover.clear()
+        sim._camera_mover = None
+
+    # Clear the vision sensor cache
+    VisionSensor.clear()
+
+    # Clear all global update states
+    for state in sim.object_state_types_requiring_update:
+        if issubclass(state, GlobalUpdateStateMixin):
+            state.global_initialize()
+
+    # Clear all materials
+    MaterialPrim.clear()
+
+    if gm.ENABLE_TRANSITION_RULES:
+        # Clear all transition rules
+        TransitionRuleAPI.clear()
+
+    # Clear uniquely named items and other internal states
+    clear_python_utils()
+    clear_usd_utils()
+
+    # Clear some internals here.
+    sim._objects_to_initialize = []
+    sim._objects_require_contact_callback = False
+    sim._objects_require_joint_break_callback = False
+    sim._link_id_to_objects = dict()
+    sim._callbacks_on_play = dict()
+    sim._callbacks_on_stop = dict()
+    sim._callbacks_on_import_obj = dict()
+    sim._callbacks_on_remove_obj = dict()
+
```

Comment:
The whole reason we moved this here is so that we can just remove the stage in Omni and let Simulator create a new one. So let's not forget to do that! Feel free to ask questions on Slack if the whole thing is not clear.